pjsip/build/Makefile

```diff
@@ -271,15 +271,15 @@ $(TEST_EXE): $(PJSIP_UA_LIB) $(PJSIP_UA_SONAME)
 $(TEST_EXE): $(PJSIP_SIMPLE_LIB) $(PJSIP_SIMPLE_SONAME)
 $(TEST_EXE): $(PJSUA_LIB_LIB) $(PJSUA_LIB_SONAME)
 $(TEST_EXE):
-	$(MAKE) -f $(RULES_MAK) APP=TEST app=pjsip-test $(subst /,$(HOST_PSEP),$(BINDIR)/$@)
+	CFLAGS="$(CFLAGS) $(SDL_CFLAGS)" $(MAKE) -f $(RULES_MAK) APP=TEST app=pjsip-test $(subst /,$(HOST_PSEP),$(BINDIR)/$@)
```

Comment:
shouldn't SDL_CFLAGS already be included in PJ_VIDEO_CFLAGS?

In any case, the addition of flags for such specific library is better not to be put here. Same for below, as well as for Samples.mak above (find out why the existing PJ_VIDEO_CFLAGS is not working and how to use it, if possible).

Reply:
Yes I see your point and I'll comment a bit on this part. Thing is, SDL on windows has a specific peculiarity that "main" needs to be redefined to "SDL_main" for the SDL library to get its necessary hook for some internal initialization and finalization (in its "substitute" "main"). To achieve that, SDL usually pushes "-Dmain=SDL_main" into its cflags (See e.g. sdl2.pc) and it is then supposed to be picked up by an application. Now because in pjproject most applications/examples do not apparently suppose any sort of SDL playback (on windows anyway), I felt a bit reluctant to somehow get all the execatables affected globally by this messy SDL hackery and tried to only fix those specific executables which needed it. Maybe it could be somehow arranged better but I'm not exactly sure yet. Please let me know what you think.

Reply:
I don't know a solution for this either. All I know is that it seems out of place.

Reply:
I was thinking about maybe introducing some additional PJ_SDL_EXE_EXTRA specifically for this. But looking at root build.mak again it seems APP_CFLAGS and APP_CXXFLAGS should already include the necessary SDL_CFLAGS. Unless they are redefined elsewhere or ignored. The variables definition and parsing system seems complicated and flexible already, I'd suppose it could allow fixing this SDL thing in a better way, but I need some hints.

Reply:
After looking at it again, I believe the modification should be as follows instead:
`export PJSUA2_TEST_CXXFLAGS = $(_CXXFLAGS) $(PJSUA2_LIB_CFLAGS)  $(PJ_VIDEO_CFLAGS)`
and
`export TEST_CFLAGS += $(_CFLAGS) $(PJ_VIDEO_CFLAGS)`

And for the one in `Samples.mak`:
```
export _CFLAGS 	:= $(PJ_CFLAGS) $(CFLAGS) $(PJ_VIDEO_CFLAGS)
export _CXXFLAGS:= $(PJ_CXXFLAGS) $(CFLAGS) $(PJ_VIDEO_CFLAGS)
```

Test it there and let me know if it works.

Reply:
Yes it does! Thank you!
Now should I push such corrections to this PR myself?
(Please excuse my ignorance, I'm not very familiar with github's workflow)

Reply:
Yes please.

Reply:
Corrections have been pushed successfully.